scripts/build.sh

```diff
@@ -0,0 +1,554 @@
+#!/bin/bash
```

Comment:
I still need to go read through this file more thoroughly (overall seems ok, but I think there are 2-3 places where script could be a bit simplified), but for a start: ShellCheck found several issues in this file, so I suggest installing it locally and fixing them (most issues are trivial).

I need to add shellcheck for bash scripts and pylint for python scripts in the future...